doc/man1/openssl-ecparam.pod.in

```diff
@@ -88,7 +88,8 @@ to get a list of all currently implemented EC parameters.
 
 =item B<-list_curves>
 
-Print out a list of all currently implemented EC parameters names and exit.
+Print out a list of 'short names' of all currently implemented EC with
+parameters and exit.
```

Comment:
I'm not sure this is really accurate.  Currently openssl ecparam --list-curves just fetches the list of built in curves and prints them along with their names.  Theres no guarantee that any of those curves will accept a given parameter, or any parameters.

I think the initial text was (at least meant to be) correct, and the intent here was for -list_curves was to print out a list of available curves, along with the textual parameter names they support (i.e. the values taken from https://www.openssl.org/docs/man3.3/man7/EVP_PKEY-EC.html).

Not sure if thats something you're interested in tackling, but (I think) that the text should be left as is and a separate issue opened to track the correction of that).

alternatively, the docs for -list_curves could be updated to just reflect that it emits a list of available built in curves, and we can use a separate issue to introduce some new option like -list_params for a given curve

I'm ok with either approach

Reply:
> Currently openssl ecparam --list-curves just fetches the list of built in curves and prints them along with their names.

Are you sure?  Have you tested it?  Does it print any curve that does not require a parameter for you, such as 25519?

Reply:
> Currently openssl ecparam --list-curves just fetches the list of built in curves and prints them along with their names.

The original text was "Print out a list of all currently implemented EC parameters names and exit."  Does `-list_curves` indeed print any EC parameters for you?  Can you show an example?

Reply:
I have, and you're right, all the curves listed accept parameters.  My point however is that it does so by happenstance, not by any direct check.  -list_curves calls EC_get_builtin_curves to get a set of, as the name implies, curves builtin to the library.  That list is statically defined and just happens to contain curves that all accept parameters.

Reply:
IMO it is misleading to say `EC with parameters`. I would keep the original text or maybe say:

`Print out a list of all EC parameter names implemented by the OpenSSL default provider and exit.`

Reply:
The original text isn't really correct either though.  it says:
"Print out a list of all currently implemented EC parameters names and exit"
but openssl ecparams doesn't do that, it just prints a list of built in curves.

I think the right thing to do is either:
a) do exactlly what you suggest  @t8m , and clarify that it just prints a list of built in curve names
or (possibly and)
b) Add a new option -list_params, to print a list of potentially accepted parameter names, independent of the curves (ideally print a list of parameters that are accepted by each curve, but I don't think thats currently possible)